lib/ansible/plugins/connection/ssh.py

```diff
@@ -307,6 +307,57 @@ class AnsibleControlPersistBrokenPipeError(AnsibleError):
     pass
 
 
+class AnsiblePasswordAuthenticationFailure(AnsibleError):
+    pass
+
+
+def _handle_error(remaining_retries, command, return_tuple, no_log, host, display=display):
+
+    # sshpass errors
+    if command == b'sshpass':
+        if return_tuple[0] == 5:
+            msg = 'Invalid/incorrect password. Skipping remaining {0} retries to prevent account lockout:'.format(remaining_retries)
+            if remaining_retries <= 0:
+                msg = 'Invalid/incorrect password:'
+            if no_log:
+                msg = '{0} <error censored due to no log>'.format(msg)
+            else:
+                msg = '{0} {1}'.format(msg, to_native(return_tuple[2].rstrip()))
+            raise AnsiblePasswordAuthenticationFailure(msg)
+        # sshpass returns codes are 1-6. We handle 5 previously, so this catches other scenarios.
+        elif return_tuple[0] in [1, 2, 3, 4, 6]:
```

Comment:
minor bikeshed: might be more effective to do a range check? >= 1 and <= 6? since 5 has already been taken care of

Reply:
The equivalent statement using comparsions seems a bit harder to read and understand:

```python
elif 1 <= return_tuple[0] < 5 or return_tuple[0] == 6:
    ...
```

Microperformance testing with `timeit` shows the comparison as slightly faster, but think the easier to read/maintain syntax makes it worth the very minor performance difference.

Reply:
`elif 1 <= return_tuple[0] <= 6:` the 5 was already handled, but yeah, as i said, not a blocker this is tiny possible perf gain.